101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1112 @@
+            "type": "string",
+            "defaultValue": "Standard_A2_v2",
+            "allowedValues": [
+                "Standard_A2_v2",
+                "Standard_A4_v2",
+                "Standard_A8_v2",
+                "Standard_D1_v2",
+                "Standard_D2_v2",
+                "Standard_D3_v2",                
+                "Standard_D2_v3",
+                "Standard_D4_v3"
+            ],
+            "metadata": {
+                "description": "What is the VM size for all VMs?"
+            }            
+        },
+        "vmStorageSkuType": {
+            "type": "string",
+            "defaultValue": "Standard_LRS",
+            "allowedValues": [
+                "StandardSSD_LRS",
+                "Standard_LRS",
+                "Premium_LRS"
+            ],
+            "metadata": {
+                "description": "What is the SKU for the storage to VM managed disks?"
+            }            
+        },
+        "adDomainName": {
+            "type": "string",
+            "defaultValue": "contoso.com",
+            "metadata": {
+                "description": "What is the new forest/root Active Directory domain name?"
+            }            
+        },
+        "vNetPrefix": {
+            "type": "string",
+            "defaultValue": "10.100",
+            "metadata": {
+                "description": "What is the prefix for the vnet and first subnet?"
+            }            
+        },
+        "vNetAddressSpace": {
+            "type": "string",
+            "defaultValue": "[concat(parameters('vNetPrefix'),'.0.0/16')]",
+            "metadata": {
+                "description": "What is the vnet address space?"
+            }            
+        },
+        "vNetSubnetAddress": {
+            "type": "string",
+            "defaultValue": "[concat(parameters('vNetPrefix'),'.0.0/24')]",
+            "metadata": {
+                "description": "What is the subnet address prefix?"
+            }            
+        },       
+        "_artifactsLocation": {
+            "type": "string",
+            "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-rds-deployment-full-ha/",
+            "metadata": {
+                "description": "Location of all scripts and DSC resources for RDS deployment."
+            }
+        },
+        "_artifactsLocationSasToken": {
+            "type": "securestring",
+            "defaultValue": "",
+            "metadata": {
+                "description": "SAS storage token to access _artifactsLocation. No need to change unless you copy or fork this template."
+            }
+        }
+    },
+    "variables": {
+        "dnsEntry": "remoteapps",
+        "brokerName": "broker",   
+        "externalFqdn": "[concat(variables('dnsEntry'),'.',parameters('externalDnsZone'))]",
+        "brokerFqdn": "[concat(variables('brokerName'),'.',parameters('externalDnsZone'))]",
+        "vmNames": [

```

Comment:
you still have the problem that these are used for things that need to be globally unique and they are not - that needs to be addressed

Reply:
Those variables are only used when deploying in "HA mode", where you need to specify an unique external dns zone owned by you. Domain dns zones are unique, allowing single owner. Example: if you own a domain named example.com, the deployment will use broker.example.com and remoteapps.example.com. You'll need to create DNS CNAMEs pointing to the load balancer FQDN, as explained in README.

Reply:
The vmNames are derived from projectName which is user supplied. Two users could use a project name of "foo".  That project name is used in the publicIpDNS and sqlServer properties that need to be globally unique and they are not currently.  So things that need to be globally unique cannot use the projectname or vmname as the only differentiator

Reply:
Ok, I was thinking you were talking about external domain names. I've made changes to vmNames, so it becomes unique. Also, set the projectName maxLength to 8, which will sum up to 5 chars from 'uniqueName' variables, plus 2 chars for vm name suffix, total 15 chars, max allowed for Windows computerNames.

Reply:
I've also made sql server name unique.